hack/e2e.sh

```diff
@@ -26,6 +26,16 @@ cd $ROOT
 
 source "${ROOT}/hack/lib.sh"
 
+# check bash version
+BASH_MAJOR_VERSION=$(echo "$BASH_VERSION" | cut -d '.' -f 1)
+# we need bash version >= 4
```

Comment:
Could we merge these 2 notes into on note?

Reply:
The first one explain the usage of the following "block" and the second one explain the usage of the following "statement".

But it is not that obvious so I think it is ok to merge.